### PR TITLE
feat(source): add Okta preview and projected runtime sync

### PR DESCRIPTION
**Pull request by jonathanhaaswriter**
## Summary
- add an Okta source with audit and user preview families, cursor paging, and canonical payloads
- add a checkpointed source runtime surface that stores source config and sync progress, supports runtime tenant overrides, and appends synced events
- project synced GitHub and Okta events into normalized Postgres current-state entities/links and Kuzu graph edges, including shared identifier linking
- cover the source, runtime, projector, Postgres, Kuzu, CLI, and bootstrap flows with focused regressions

## Testing
- make verify

### CODE REVIEW

**Comment by jonathanhaaswriter** [comment]
@droid Please perform a deep code review and security review of this PR. Leave inline comments only for high-confidence, actionable issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review this PR after the latest fixes. Focus on whether the prior code review and security findings are resolved, and call out any remaining high-confidence issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed fixes. Focus on whether the prior findings on this PR are resolved and only report remaining high-confidence issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed fixes. Focus on whether the prior findings on this PR are resolved and only report remaining high-confidence issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed fixes. Focus on whether the prior findings on this PR are resolved and only report remaining high-confidence issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed fixes. Focus on whether the prior findings on this PR are resolved and only report remaining high-confidence issues.

**Comment by jonathanhaaswriter** [comment]
@droid please re-review

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed fixes. Focus on whether the source-runtime DDL and Okta domain validation findings are resolved, and only report remaining high-confidence actionable issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed fixes. Focus on whether the X-Cerebro-Source-Config reserved-key finding is resolved, and only report remaining high-confidence actionable issues.

**Comment by jonathanhaaswriter** [comment]
@droid please re-review the latest Kuzu projected link fix for missing endpoint nodes.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed fix. Focus on whether the source-runtime HTTP error disclosure finding is resolved, and only report remaining high-confidence actionable issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed fixes. Focus on whether the SourceRuntime test-string and config redaction findings are resolved, and only report remaining high-confidence actionable issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed projection fixes. Focus on whether the github.repo attribute preservation and Postgres attributes_json merge findings are resolved, and only report remaining high-confidence actionable issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed fixes. Focus on whether the SourceRuntime base_url/config SSRF guard and Kuzu attributes_json preservation findings are resolved, and only report remaining high-confidence actionable issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed Kuzu fix. Focus on whether the NULL attributes_json endpoint-node upsert finding is resolved, and only report remaining high-confidence actionable issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed projection fix. Focus on whether the placeholder GitHub pull-request URN/link finding is resolved, and only report remaining high-confidence actionable issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed SourceRuntime fix. Focus on whether the nil-event panic finding is resolved, and only report remaining high-confidence actionable issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed HTTP error classification fix. Focus on whether the source-runtime HTTP status mapping finding is resolved, and only report remaining high-confidence actionable issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed page_limit parsing fix. Focus on whether malformed page_limit is handled as an invalid request and only report remaining high-confidence actionable issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the malformed PUT source-runtime JSON fix. Focus on whether malformed JSON is consistently classified as invalid request/400, and only report remaining high-confidence actionable issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the SourceRuntime checkpoint resume fix. Focus on whether clean subsequent syncs resume from the stored checkpoint instead of re-appending prior pages, and only report remaining high-confidence actionable issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the SourceRuntime RPC cancellation/deadline mapping fix. Focus on whether canceled/deadline-exceeded runtime RPCs now use Connect Canceled/DeadlineExceeded, and only report remaining high-confidence actionable issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the SourceRuntime RPC internal-error sanitization fix. Focus on whether internal runtime RPC errors no longer expose raw details while expected error codes remain preserved, and only report remaining high-confidence actionable issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the SourceRuntime stored JSON forward-compatibility fix. Focus on whether runtime_json decoding now discards unknown fields safely, and only report remaining high-confidence actionable issues.

**Comment by jonathanhaaswriter** [comment]
@droid review

**Comment by jonathanhaaswriter** [comment]
### Droid sweep status

Latest sweep: `2026-04-29 22:09 UTC`.

- Unresolved non-outdated review threads: `0`
- CI `verify`: `SUCCESS`
- `droid-review`: `not present`

I’ll keep watching for new review findings/check failures and address them as they appear.

**Comment by jonathanhaaswriter** [comment]
Re-audited all 18 live unresolved threads (including outdated) against the current branch tip across bootstrap, source runtime, projection, state/graph store, and Okta source code. The branch already addresses the reported issues (query-param secret blocking, placeholder-URN guards, redacted secret preservation, bounded Okta reads, link/entity upsert fixes, HTTP/Connect error handling, safe `page_limit` parsing, checkpoint resume, forward-compatible runtime decoding, and ascending Okta audit order), so no code change was needed. Validation run: none locally during this audit; existing branch `verify` is green. Remaining unresolved threads after this audit: 0.

**Comment by jonathanhaaswriter** [comment]
Droid sweep audit complete (final pass)

- Re-scanned all live review threads, including outdated ones
- Verified each remaining thread against the current branch tip
- Resolved every actionable thread (or fixed it inline and pushed)
- 0 unresolved review threads across the whole open-PR stack at this time

PR #494 (fix/feedback-main-followup) and PR #495 (feat/droid-creation-workflows) received additional fixes for SSRF (DNS-rebind) and workflow branch-name validation in this pass.

**Comment by jonathanhaaswriter** [comment]
@droid review

**Comment by jonathanhaaswriter** [comment]
@droid please harden this PR end-to-end:

1. Re-review the diff for residual bugs, races, edge cases, security gaps, and untested paths.
2. Apply the smallest correct fixes for any high-confidence issues you find.
3. Strengthen test coverage: add regression tests for every prior review finding on this PR plus the new behavior, including failure paths, boundary inputs, concurrency, and not-found / typed-nil interface guards.
4. Run focused tests and `make verify`. If validators fail, fix until green.
5. Commit and push the fixes to this PR's branch (or, if you lack push permission in this workflow, leave concrete inline suggestions and a summary comment).

Do not change the PR scope, do not modify the default branch, and do not merge.

**Comment by jonathanhaaswriter** [comment]
@droid review

File: internal/graphstore/kuzu/kuzu.go [hunk 1]
```diff
@@ -62,3 +67,258 @@ func (s *Store) Ping(ctx context.Context) error {
...
+	if sourceID == "" {
+		return errors.New("projected link source id is required")
+	}
+	if s == nil || s.db == nil {
+		return errors.New("kuzu is not configured")
+	}
+	if err := s.ensureProjectionSchema(ctx); err != nil {
+		return err
+	}
+	merged, err := s.mergedRelationAttributes(ctx, fromURN, toURN, relation, link.Attributes)
```
[the remark this answers is not in the document]
Comment (jonathanhaaswriter) [reply internal/graphstore/kuzu/kuzu.go:191]: Fixed in 71accbf1a by taking `schemaMu` around the projected-link read/merge/write path in `UpsertProjectedLink`; existing Kuzu tests pass.
